autopilot/manager.go

```diff
@@ -287,3 +287,35 @@ func (m *Manager) QueryHeuristics(nodes []NodeID) (HeuristicScores, error) {
 	log.Debugf("Querying heuristics for %d nodes", len(n))
 	return m.pilot.queryHeuristics(n)
 }
+
+// SetNodeScores is used to set the scores of the given heuristic, if it is
+// active, and ScoreSettable.
+func (m *Manager) SetNodeScores(name string, scores map[NodeID]float64) error {
+	// Check if the targeted heuristc is among the active sub-heuristics.
+	found := false
+	for _, h := range m.cfg.PilotCfg.Heuristic.SubHeuristics() {
```

Comment:
Intentionally left this as an array to support the case with duplicate sub-heuristics with the same name.

Reply:
not a blocker, but having trouble determining when this would be useful. if multiple subheurisitcs share the same name and are the same object, we only need to invoke `SetNodeScores` once. If they are distinct objects, it's ambiguous which one is being targeted (I would think this should produce an error when creating the manager) and may result in undefined behavior. In the case of `WeightedCombAttach`, if two distinct heuristics both share the name `externalattach` with weights 0.25 and 0.5, they could equally be represented as a single heuristic w/ weight 0.75.